resources/js/Components/UserActivityMap.tsx

```diff
@@ -0,0 +1,371 @@
+import axios from "axios";
+import { useEffect, useState, useRef } from "react";
+import { DropdownControl } from "./inputs";
+import { User } from "@/common";
+
+type ActivityMapData = {
+    user_id: number;
+    date: string;
+    active_course_count: string;
+    total_activity_time: string;
+    total_activity_time_quartile: number;
+};
+
+/* interface for dynamically generated year options */
+interface ValidYears {
+    [year: string]: string;
+}
+
+/* subtract a year from a date */
+const subtractYear = (date: Date) => {
+    const newDate = new Date(date);
+    newDate.setUTCFullYear(date.getUTCFullYear() - 1);
+    newDate.setUTCDate(newDate.getUTCDate() + 1);
+    return newDate;
+};
+
+/* predefining tailwind strings lets us assign them dynamically later */
+const quartileColors: string[] = [
+    "bg-quartile-0",
+    "bg-quartile-1",
+    "bg-quartile-2",
+    "bg-quartile-3",
+    "bg-quartile-4",
+];
+
+/* node styles dependant on size state breakpoints */
+const nodeSizes: string[] = [
+    "w-2 h-2 text-xs rounded-sm",
+    "w-3 h-3 text-xs rounded-sm",
+    "w-4 h-4 text-sm rounded-md",
+];
+
+/* spacer widths based on size state breakpoints */
+const spacerWidths: string[] = ["w-[294px]", "w-[516px]", "w-[714px]"];
+
+/* main component for the user activity map */
+export default function UserActivityMap({
+    user,
+    size = 1,
+}: {
+    user: User;
+    size?: number;
+}) {
+    const userCreatedAt = user.created_at
+        ? new Date(user.created_at)
+        : new Date(2024, 0, 1);
+
+    const [mapSize, setMapSize] = useState(size);
+    const [yearEnd, setYearEnd] = useState(new Date());
+    const [dropdownValDesc, setDropdownValDesc] = useState("the past year");
+    const [activityTime, setActivityTime] = useState("");
+    const containerRef = useRef<HTMLDivElement>(null);
+
+    const generateYearOptions = () => {
+        let years: ValidYears = { "Past year": "Past year" };
+        const currentYear = new Date().getUTCFullYear();
+        let i = currentYear - userCreatedAt.getUTCFullYear();
+        for (i; i >= 0; i--) {
+            /* added " " to change listed order without impacting appearance */
+            let str = " " + String(userCreatedAt.getUTCFullYear() + i);
+            years[str] = str;
+        }
+        return years;
+    };
+
+    const fetchActivityData = async () => {
+        const startDate = new Date(subtractYear(yearEnd));
+        const response = await axios.get(
+            "/api/v1/user-activity-map/" +
+                user.id +
+                "?start_date=" +
+                startDate.toISOString().split("T")[0] +
+                "&end_date=" +
+                yearEnd.toISOString().split("T")[0],
+        );
+        const data = await response.data;
+        setActivityData(data);
+    };
```

Comment:
Rather than use axios.get, we have been using a react hook to fetch data called useSWR (https://swr.vercel.app/). You can see an example in `Users.tsx` (and other files like provider platforms, user activity, etc). With this you can also handle loading & errors and render something accordingly.
This is totally my bad because I completely forgot to add this into the `CONTRIBUTING.md` file. I'll do that soon :)